src/Datadog.Trace/ExtensionMethods/RandomExtensions.cs

```diff
@@ -4,6 +4,42 @@ namespace Datadog.Trace.ExtensionMethods
 {
     internal static class RandomExtensions
     {
+#if !NETSTANDARD
+        private static readonly Random GlobalRandom = new Random();
+#endif
+
+        [ThreadStatic]
+        private static Random _threadRandom;
+
+        public static Random GetRandom()
+        {
+            var random = _threadRandom;
+
+            if (random == null)
+            {
+#if NETSTANDARD
+                random = new Random();
+#else
+                // On .net framework, the clock is used to seed the new random instances
+                // Resolution is too low, so if a bunch of threads are created at the same time,
+                // they'll all get the same seed.
+                // Instead, use a shared random instance to generate the seed
+                int seed;
+
+                lock (GlobalRandom)
+                {
```

Comment:
You take a lock because Next is not thread-safe, makes sense. Now since you already took it, you might as well check for the thread local still being null

```
Random random = _threadRandom;

if (random != null)
{
    return random;
}

lock (GlobalRandom)
{
    random = _threadRandom;
    if (random != null)
    {
        int seed = GlobalRandom.Next();
        random = new Random(seed);
         _threadRandom = random;
    }
}

return random;
```

You hold the lock for longer the first time and for shorter the second. 
either way, this is minor, as this happens only once per thread. :)

Reply:
Checking again does not make sense. The field is thread-static, the value cannot possibly change without the current thread changing it